hal/spm_api.h

```diff
@@ -0,0 +1,103 @@
+/* Copyright (c) 2017-2018 ARM Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __SPM_API_H__
+#define __SPM_API_H__
+
+
+/** @addtogroup SPM
+ * The SPM (Secure Partition Manager) is responsible for isolating software in
+ * partitions, managing the execution of software within partitions, and
+ * providing IPC between partitions.
+ * @{
+ */
+
+#include <stdint.h>
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/** @addtogroup HAL-SPE
```

Comment:
These docs won't be generated by doxygen unless you ad the COMPONENTS_SPE flag to: `doxyfile_options` and `doxygen_options.json`